plume-front/src/editor.rs

```diff
@@ -62,7 +69,183 @@ impl From<stdweb::private::ConversionError> for EditorError {
+                            .try_into()
+                            .unwrap_or_default()
+                    } else {
+                        (js! { return @{&ch}.outerHTML; })
+                            .try_into()
+                            .unwrap_or_default()
+                    }
+                }
+                NodeType::Text => ch.node_value().unwrap_or_default(),
+                _ => unreachable!(),
+            };
+            format!("{}\n\n{}", md, to_append)
+        })
+    }
+}
+fn get_subtitle() -> String {
+    if is_basic_editor() {
+        let subtitle: InputElement =
+            InputElement::try_from(document().get_element_by_id("subtitle").unwrap())
+                .ok()
+                .unwrap();
+        console!(log, "Got subtitle");
+        subtitle.raw_value()
+    } else {
+        let subtitle_element = HtmlElement::try_from(
+            document()
+                .query_selector("#plume-editor > h2")
+                .unwrap()
+                .unwrap(),
+        )
+        .ok()
+        .unwrap();
+        subtitle_element.inner_text()
+    }
+}
+fn get_tags() -> String {
+    let tags: InputElement = InputElement::try_from(document().get_element_by_id("tags").unwrap())
+        .ok()
+        .unwrap();
+    tags.raw_value()
+}
+fn get_license() -> String {
+    let license: InputElement =
+        InputElement::try_from(document().get_element_by_id("license").unwrap())
+            .ok()
+            .unwrap();
+    console!(log, "Got license");
+    license.raw_value()
+}
+fn get_cover() -> String {
+    let cover: SelectElement =
+        SelectElement::try_from(document().get_element_by_id("cover").unwrap())
+            .ok()
+            .unwrap();
+    console!(log, "Got cover");
+    cover.raw_value()
+}
+fn autosave() {
+    let info: AutosaveInformation = AutosaveInformation {
+        contents: get_editor_contents(),
+        title: get_title(),
+        subtitle: get_subtitle(),
+        tags: get_tags(),
+        license: get_license(),
+        last_saved: Date::now(),
+        cover: get_cover(),
+    };
+    let id = get_autosave_id();
+    match window()
+        .local_storage()
+        .insert(&id, &serde_json::to_string(&info).unwrap())
+    {
+        Ok(_) => {}
+        _ => console!(log, "Autosave failed D:"),
+    }
+}
+//This is only necessary until we go to stdweb 4.20 at least
+fn confirm(message: &str) -> bool {
+    return js! {return confirm(@{message});} == true;
+}
+fn load_autosave() {
+    if let Some(autosave_str) = window().local_storage().get(&get_autosave_id()) {
+        let autosave_info: AutosaveInformation = serde_json::from_str(&autosave_str).ok().unwrap();
+        let message = format!(
+            "{} {}?",
+            i18n!(
+                CATALOG,
+                "Do you want to load the local autosave last edited at"
+            ),
+            Date::from_time(autosave_info.last_saved).to_date_string()
+        );
```

Comment:
You should use `i18n!` for formatting, to allow the date to be placed somewhere else in the sentence if translators need to do so.

```rust
i18n!(
    CATALOG,
    "Do you want to load the local autosave last edited at {}";
    Date::from_time(autosave_info.last_saved).to_date_string()
)
```

Reply:
Didn't realize it was possible to do that with `i18n!`, that's nifty :)